### PR TITLE
[ARCTIC-1440][Improvement]: Improve the unit tests for module Flink

### LINKED ISSUE #1440: [Subtask]: Improve the unit tests for module Flink
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/arctic/issues?q=is%3Aissue) and found no similar issues.

### Description

* Fix #1364 
* Integrate the new unit testing framework
* Complementary test cases for the Flink module
* Improve unit test coverage

### Parent issue

https://github.com/NetEase/arctic/issues/937

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by lklhdu**
## Why are the changes needed?
resolve #1440 
The current progress of the unit tests refactoring under the Flink module is the integration of a new unit testing framework, however, the test coverage of the module still falls short of the target of 90%.

Subsequent PRs will further improve the unit test coverage of the module.

## Brief change log

  - *Integrate the new unit testing framework under the core module*
  - *Only the flink 1.12 version is affected.*

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink/v1.12/flink/src/test/java/com/netease/arctic/flink/FlinkTestBase.java [hunk 1]
```diff
@@ -105,66 +102,38 @@ public class FlinkTestBase extends TableTestBase {
   protected Map<String, String> props;
   private volatile StreamExecutionEnvironment env = null;
 
+  public static final Schema TABLE_SCHEMA = new Schema(
```
Comment (YesOrNo828) [inline flink/v1.12/flink/src/test/java/com/netease/arctic/flink/FlinkTestBase.java:105]: I think this TABLE_SCHEMA is duplicated with BasicTableTestHelper.TABLE_SCHEMA, maybe you can fetch the schema by `BasicTableTestHelper.tableSchema()`

File: flink/v1.12/flink/src/test/java/com/netease/arctic/flink/FlinkTestBase.java [hunk 2]
```diff
@@ -105,66 +102,38 @@ public class FlinkTestBase extends TableTestBase {
...
+    Types.NestedField.required(3, "ts", Types.LongType.get()),
+    Types.NestedField.required(4, "op_time", Types.TimestampType.withoutZone())
+  );
+
   public static final TableSchema FLINK_SCHEMA = TableSchema.builder()
-      .field("id", DataTypes.INT())
-      .field("name", DataTypes.STRING())
-      .field("op_time", DataTypes.TIMESTAMP())
-      .build();
+    .field("id", DataTypes.INT())
```
Comment (YesOrNo828) [inline flink/v1.12/flink/src/test/java/com/netease/arctic/flink/FlinkTestBase.java:113]: You can also have a subclass extending `BasicTableTestHelper` which can have its own TABLE_SCHEMA so you can keep the same schema as before.

File: flink/v1.12/flink/src/test/java/com/netease/arctic/flink/catalog/TestCatalog.java [hunk 3]
```diff
@@ -18,65 +18,70 @@
...
-    super.config(TEST_CATALOG_NAME);
+  private volatile StreamExecutionEnvironment env = null;
+  private volatile StreamTableEnvironment tEnv = null;
+
+  public TestCatalog() {
+    super(new BasicCatalogTestHelper(TableFormat.MIXED_ICEBERG));
   }
 
-  @Test
-  public void testRefresh() {
```
Comment (YesOrNo828) [inline flink/v1.12/flink/src/test/java/com/netease/arctic/flink/catalog/TestCatalog.java:56]: Why is this method discarded?

File: flink/v1.12/flink/src/test/java/com/netease/arctic/flink/table/TestKeyed.java [hunk 4]
```diff
@@ -165,14 +219,31 @@ private void prepareLog() {
...
-    if (isHive) {
-      hiveTableTestBase.clearTable();
-    }
     if (Objects.equals(logType, LOG_STORE_STORAGE_TYPE_PULSAR)) {
       pulsarHelper.op().deleteTopicByForce(topic);
     }
   }
 
+  @Test
+  public void testRefresh() {
```
Comment (YesOrNo828) [inline flink/v1.12/flink/src/test/java/com/netease/arctic/flink/table/TestKeyed.java:228]: Why do you move the testRefresh() here? WDYT?

**Review by YesOrNo828 — COMMENTED** [review]
@lklhdu I left some comments.

File: flink/v1.12/flink/src/test/java/com/netease/arctic/flink/catalog/TestCatalog.java (hunk 3 above)
Comment (lklhdu) [reply flink/v1.12/flink/src/test/java/com/netease/arctic/flink/catalog/TestCatalog.java:56]: I have moved the testRefresh() to a new TestClass named TestTableRefresh.

File: flink/v1.12/flink/src/test/java/com/netease/arctic/flink/FlinkTestBase.java (hunk 2 above)
Comment (lklhdu) [reply flink/v1.12/flink/src/test/java/com/netease/arctic/flink/FlinkTestBase.java:113]: thanks, now use BasicTableTestHelper.TABLE_SCHEMA can be aligned with other modules.

**Pull request merged**